odoo/models.py

```diff
@@ -4134,21 +4262,27 @@ def _write(self, vals):
             vals.setdefault('write_date', self.env.cr.now())
 
         # determine SQL values
-        columns = {}                    # {column_name: value}
+        columns = []
+        params = []
 
         for name, val in sorted(vals.items()):
             if self._log_access and name in LOG_ACCESS_COLUMNS and not val:
                 continue
             field = self._fields[name]
             assert field.store
             assert field.column_type
-            columns[name] = val
+            if field.translate is True and val:
+                columns.append(f'"{name}" = %s || COALESCE("{name}", \'{{}}\'::jsonb) || %s')
+                params.append(Json({} if 'en_US' in val.adapted else {'en_US': next(iter(val.adapted.values()))}))
+                params.append(val)
```

Comment:
This definitely needs a comment / explanation.

My understanding is that the first segment / param is to set the new value as `en_US` if there's no existing value there? Then merge in the current values, and finally the values being explicitely provided / set?

Reply:
Also in case there can be mutiple keys in `val`, order is not round-tripped through postgres, which might lead to odd situations. Though I guess this would only fire on the first write to a translatable field.